### PR TITLE
Add MobileCore.getApplication API

### PR DESCRIPTION
**Pull request by kevinlind**
Adding documentation for new MobileCore.getApplication API. This is an Android only API.

### CODE REVIEW

File: resources/building-mobile-extensions/get-the-application-context.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Get the Application Context
+
+In Android, you can retrieve the `android.app.Application` instance using the `MobileCore.getApplication()` API. The returned Application represents the base class which maintains the global application state. For more information, [Moble Core API Reference](../../using-mobile-extensions/mobile-core/configuration-reference/mobile-core-api-reference.md)
```
Comment (rekrishn) [inline resources/building-mobile-extensions/get-the-application-context.md]: - In the 2nd sentence, does "Application" have to be capitalized? if not, please change to a lower case "a". 
- In the 3rd sentence, please add "see" after the comma.

File: resources/building-mobile-extensions/get-the-application-context.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+{% tabs %}
+{% tab title="Android" %}
+
+### Android
+
+### getApplication
+
+{% hint style="warning" %}
+
+`MobileCore.getApplication` may return `null` if the Application object was destroyed or if `MobileCore.setApplication`was not previously called.
```
Comment (rekrishn) [inline resources/building-mobile-extensions/get-the-application-context.md]: - Please change "may" to "might".
- Does "Application" have to be capitalized? If not, please change to a lower case "a".

File: using-mobile-extensions/mobile-core/configuration-reference/mobile-core-api-reference.md [hunk 3]
```diff
@@ -1,5 +1,82 @@
 # Mobile Core API Reference
 
+# Application reference
+
+When building Android applications, it is required to pass the `android.app.Application` reference to the Mobile SDK, which allows the Mobile SDK to access the `android.app.Context` and monitor the lifecycle of the Andorid application.
```
Comment (rekrishn) [inline using-mobile-extensions/mobile-core/configuration-reference/mobile-core-api-reference.md]: Please replace "it is required to" with "the `android.app.Application` reference must be passed to..."

File: using-mobile-extensions/mobile-core/configuration-reference/mobile-core-api-reference.md [hunk 4]
```diff
@@ -1,5 +1,82 @@
...
+
+{% hint style="warning" %}
+
+Android applications must call MobileCore.setApplication() before calling any other Mobile SDK API.
```
Comment (rekrishn) [inline using-mobile-extensions/mobile-core/configuration-reference/mobile-core-api-reference.md]: Please tag "MobileCore.setApplication()" as code.

File: using-mobile-extensions/mobile-core/configuration-reference/mobile-core-api-reference.md [hunk 5]
```diff
@@ -1,5 +1,82 @@
...
+
+{% tab title="Android" %}
+
+#### Java
+
+### getApplication
+
+{% hint style="warning" %}
+
+`MobileCore.getApplication` may return `null` if the Application object was destroyed or if `MobileCore.setApplication` was not previously called.
```
Comment (rekrishn) [inline using-mobile-extensions/mobile-core/configuration-reference/mobile-core-api-reference.md]: Please change "may" to "might".

**Comment by kevinlind** [comment]
Closing as it is based on the staging branch.
Opened PR #64 which is identical to this PR except it is based off the master branch.

**Pull request closed without merging**